docs/source/checkpoints/dist_ckpt.rst

```diff
@@ -0,0 +1,421 @@
+6. All ShardedTensors are saved.
+7. The ``metadata.json`` file with backend and version metadata is saved to the checkpoint directory.
+
+dist_checkpointing.load
+^^^^^^^^^^^^^^^^^^^^^^^
+The ``dist_checkpointing.load`` function is the main entry point for checkpoint loading.
+It requires providing a sharded state dict (in order to implicitly define mappings between local tensors and checkpoint tensors) and loading strategies.
+In practice, the same sharded state dict can be usually used for both saving and loading (the sharded state dict for loading will just contain tensors with uninitialized data).
+
+When the sharded state dict is provided as input, it is processed in the following way (see also ``load`` function `documentation <https://docs.nvidia.com/megatron-core/developer-guide/latest/api-guide/dist_checkpointing.html#module-core.dist_checkpointing.serialization>`_):
+
+1. The "common" state dict is loaded from the checkpoint. This forms the base of the resulting state dict.
+2. The ShardedTensorFactories from the input sharded state dict are applied.
+3. The LocalNonPersistentObjects are extracted from the input sharded state dict, unwrapped and added to the resulting state dict.
+4. The ShardedObjects are extracted and loaded from the checkpoint into the resulting state dict.
+5. The ShardedTensors are extracted and loaded from the checkpoint into the resulting state dict.
+6. Factory merges are applied (see `Optimizers`_ for explanation).
+
+This results in a *regular* state dict with plain tensors that can be further processed by the application (which usually means running ``model.load_state_dict(state_dict)``.
+
+
+dist_checkpointing.load_common_state_dict
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+The ``dist_checkpointing.load_common_state_dict`` function is an entry point that allows loading only the “common” part of the checkpoints.
+Most of the checkpoint config and metadata can be loaded with this method, which allows skipping data loading in order to take decisions regarding checkpoint config, version, etc.
+
+dist_checkpointing.load_tensors_metadata
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+The ``dist_checkpointing.load_tensors_metadata`` function is an entry point that allows reading all ShardedTensors metadata from the checkpoint without loading any data.
+The result is a sharded state dict with trivial sharding (every tensor is sharded into one big shard).
+
+dist_checkpointing.load_plain_tensors
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+The ``dist_checkpointing.load_plain_tensors`` function is an entry point that allows reading sharded tensors stored in the checkpoint without any sharding (as plain tensors).
+This function is simply a composition of ``load_tensors_metadata`` and ``save``.
+
+Save and Load Strategies
+------------------------
+There are multiple ways to save a sharded state dict into a serialized checkpoint. They can be provided by the user as saving and loading strategies.
```

Comment:
"They can be provided by the user as saving and loading strategies." this sentence is confusing. You mean something like the objects need to be specified by the user?